Lib/distutils/ccompiler.py

```diff
@@ -8,7 +8,7 @@
 from distutils.spawn import spawn
 from distutils.file_util import move_file
 from distutils.dir_util import mkpath
-from distutils.dep_util import newer_pairwise, newer_group
```

Comment:
Is this function used at all? In any case left this on distutils experts. This is very old and stable code. Maybe it is a bug that newer_pairwise is not used.

Reply:
It's part of the public API:
https://docs.python.org/dev/distutils/apiref.html#distutils.dep_util.newer_pairwise

> This is very old and stable code. Maybe it is a bug that newer_pairwise is not used.

An unused import is a bug, I don't think that not using a function is a bug :-) The code uses newer_group().

Reply:
Yes, but maybe newer_pairwise() should be used instead of newer_group(). Or there is a typo in the name of newer_pairwise, and this is why it can't be searched in the code. Let distutils experts to review this change.

Reply:
> Yes, but maybe newer_pairwise() should be used instead of newer_group(). Or there is a typo in the name of newer_pairwise, and this is why it can't be searched in the code. Let distutils experts to review this change.

I looked at the code, and I don't see how it could be a "typo". newer_pairwise() expects a list of objects in the second argument, whereas newer_group() expects a single "target" as the second argument. The code looks perfectly fine:

```
    def _need_link(self, objects, output_file):
        """Return true if we need to relink the files listed in 'objects'
        to recreate 'output_file'.
        """
        if self.force:
            return True
        else:
            if self.dry_run:
                newer = newer_group (objects, output_file, missing='newer')
            else:
                newer = newer_group (objects, output_file)
            return newer
```

Reply:
Seems good.

(In any case, if it was a mistake to use newer_group, the mistake has been here for a long time.
Changing it would need its own ticket.)